lib/python/qmk/keymap.py

```diff
@@ -30,16 +56,21 @@ def template(keyboard):
     Args:
         keyboard
             The keyboard to return a template for.
+
+        experimental
+            True if the user wants to attempt compilation with encoder support
     """
     template_file = Path('keyboards/%s/templates/keymap.c' % keyboard)
 
     if template_file.exists():
         return template_file.read_text()
```

Comment:
Your test json fails, because https://github.com/qmk/qmk_firmware/blob/master/keyboards/handwired/onekey/pytest/templates/keymap.c exists and the function will return with its content instead of your template.

@skullydazed, we likely need to think it through if we want to keep template files or just use the ones defined in keymap.py. (maybe move them to a better place or even use a templating engine?)

Reply:
Moved the json to a different folder so it is forced to use the encoder default string rather than the template

Reply:
Keyboard maintainers can provide their own base templates. I don't see a need to break that right now.